src/main/java/com/tysng/duke/storage/Storage.java

```diff
@@ -0,0 +1,141 @@
+package com.tysng.duke.storage;
+
+import com.tysng.duke.domain.Deadline;
+import com.tysng.duke.domain.Event;
+import com.tysng.duke.domain.Task;
+import com.tysng.duke.domain.Todo;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class handles saving and loading from local files.
+ */
+public class Storage {
+    private static final String FILENAME = "duke.txt";
+
+    /**
+     * The platform-independent path to the /data folder.
+     */
+    private static Path DATA_FOLDER = Path.of(".", "data");
+
+    private static SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("EEE MMM dd HH:mm:ss zzz yyyy");
+
+    /**
+     * The Path object is assigned at run time.
+     */
+    private Path storageFilePath;
+
+    private Storage() {
+        this.storageFilePath = Path.of(DATA_FOLDER.toString(), FILENAME);
+    }
+
+    /**
+     * Constructs a new Storage class.  The static factory method will create the data.txt file if the path
+     * and the file does not already exist.
+     *
+     * @return the constructed class
+     */
+    public static Storage initialize() {
+        Storage storage = new Storage();
+        try {
+            if (!Files.exists(DATA_FOLDER)) {
+                Files.createDirectories(DATA_FOLDER);
+            }
+
+            if (!Files.exists(storage.storageFilePath)) {
+                Files.createFile(storage.storageFilePath);
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return storage;
+    }
+
+    /**
+     * Saves the given task list to local files.
+     *
+     * @param taskList a list of tasks
+     */
+    public void saveTasks(List<Task> taskList) {
+        List<String> entries = taskList.stream().map(this::toFileEntry).collect(Collectors.toList());
+        try {
+            Files.write(this.storageFilePath, entries);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Loads the tasks from local storage.  When the storage text file is not found, it writes the stack trace
+     * to stderr and returns an empty list.
+     *
+     * @return the loaded tasks
+     */
+    public List<Task> loadTasks() {
+        try {
+            List<String> entries = Files.readAllLines(this.storageFilePath);
+            return entries.stream().map(this::parse).flatMap(Optional::stream).collect(Collectors.toList());
+        } catch (IOException e) {
+            e.printStackTrace();
+            return List.of();
+        }
+    }
+
+    private String toFileEntry(Task input) {
```

Comment:
Storage class handles the parsing and serializing of the Task class. It takes in and spits out Task objects. Serializing is done in this layer because the Task class should not be concerned with how it is represented in the Storage (or the database, in actual Java dev).

Reply:
Makes sense. TIL.